tinygrad/runtime/ops_cpu.py

```diff
@@ -16,6 +16,10 @@ def shape_to_axis(old_shape:Tuple[int, ...], new_shape:Tuple[int, ...]) -> Tuple
   MovementOps.RESHAPE: lambda x, arg: x.reshape(arg), MovementOps.SHRINK: lambda x, arg: x[tuple(slice(p[0], p[1], None) for p in arg)],
 }
 
+def match_types(x, y):
```

Comment:
This is making many needless copies.

"By default, astype always returns a newly allocated array"

Reply:
Sorry. I changed it to use copy=False for both x and y